proxy/envoy/config.go

```diff
@@ -260,8 +261,24 @@ func buildHTTPListener(mesh *proxyconfig.ProxyMeshConfig, role proxy.Node, route
 		Config: FilterRouterConfig{},
 	})
 
+	service := ""
+	if instances != nil {
+		// join service names with a comma
+		serviceSet := make(map[string]bool, len(instances))
+		for _, instance := range instances {
+			serviceSet[instance.Service.Hostname] = true
+		}
+		services := make([]string, 0, len(serviceSet))
+		for service := range serviceSet {
+			services = append(services, service)
+		}
+
+		sort.Strings(services)
+		service = strings.Join(services, ",")
```

Comment:
@douglas-reid Costin really needs to set target service in pilot. Can we coordinate on the choice of the target service among the list of instances? Is it first by alphabetic order?

Reply:
After our offline discussion, I'm not entirely sure what the action here is. My understanding was that, for `target.service`, Mixer was going to get out of the business of trying to figure that out. If there are multiple services per pod, are we going to use an annotation to select the canonical Istio service (maybe "istio=" ?)?

Reply:
We essentially need to pass a ISTIO_SERVICE_NAME env var to pilot agent (vm or k8s).. and if this gets passed to envoy (via service cluster option), then we can infer the target service easily. This is a generic solution that is independent of the platform. the agent can default to istio-proxy if not present..

Reply:
Except the late bound service problem... Let's not bring this one up again.

Reply:
For the general case - you can have different listeners, each with a different target.service. 

There are many ideas and suggestions on how to improve this - the point of the PR is to revert to the status
before it got broken, it will not be better than what we had 2 weeks ago when it broke.

IMHO we need a proper design and discussion before making this kind of changes, there are many 
possible solutions and tradeoffs.